tracer/src/Datadog.Trace.SourceGenerators/TagsListGenerator/Sources.cs

```diff
@@ -0,0 +1,337 @@
+// <copyright file="Sources.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Collections.Generic;
+using System.Text;
+
+namespace Datadog.Trace.SourceGenerators.TagsListGenerator
+{
+    internal static class Sources
+    {
+        public const string Attributes =
+            @"// <auto-generated/>
+#nullable enable
+
+namespace Datadog.Trace.SourceGenerators;
+
+/// <summary>
+/// Used to designate a property as corresponding to the provided
+/// <see cref=""TagName""/>. Should only be used in ITags classes.
+/// Used for source generation.
+/// </summary>
+[System.AttributeUsage(System.AttributeTargets.Property, AllowMultiple = false)]
+internal sealed class TagNameAttribute : System.Attribute
```

Comment:
What do you think of calling them `[Tag]` and `[Metric]`? As attributes, they describe the property (i.e. the tag), not the attribute's ctor parameter (the tag's name).

Reply:
Makes sense, done in https://github.com/DataDog/dd-trace-dotnet/pull/2076/commits/fb5acec75a6bdc68d153f1cef31dfd29a338d297